### ISSUE TITLE
m2e generates JDT APT preferences with platform-specific paths

### ISSUE THREAD

**Issue by tomaswolf** [eclipse-m2e/m2e-core#989]
When m2e configures a project with APT enabled, it generates a JDT `AptConfig` with platform-specific paths. Example on Windows:

**`.settings/org.eclipse.jdt.apt.core.prefs`**
```
eclipse.preferences.version=1
org.eclipse.jdt.apt.aptEnabled=true
org.eclipse.jdt.apt.genSrcDir=target\\generated-sources\\annotations
org.eclipse.jdt.apt.genTestSrcDir=target\\generated-test-sources\\test-annotations
```
When these settings are checked in and then someone checks out the project on OS X (and presumably also other Unix systems), JDT will create a single folder named "target\generated-sources\annotations" instead of a folder hierarchy.

m2e should write the platform-independent path as
```
eclipse.preferences.version=1
org.eclipse.jdt.apt.aptEnabled=true
org.eclipse.jdt.apt.genSrcDir=target/generated-sources/annotations
org.eclipse.jdt.apt.genTestSrcDir=target/generated-test-sources/test-annotations
```
That will work on both Windows and Unix.

A correction seems to be needed in `org.eclipse.m2e.apt.internal.AbstractAptConfiguratorDelegate.java`.

P.S.: I can't fix this myself since the Oomph setup leaves me with a broken Eclipse workspace: "Project build error: Unresolveable build extension: Plugin org.eclipse.tycho:tycho-maven-plugin:3.0.0 or one of its dependencies could not be resolved: Failed to collect dependencies at org.eclipse.tycho:tycho-maven-plugin:jar:3.0.0 -> org.eclipse.tycho:tycho-core:jar:3.0.0 -> org.eclipse.tycho:p2-maven-plugin:jar:3.0.0 -> org.eclipse.platform:org.eclipse.equinox.p2.core:jar:2.9.200 -> org.eclipse.platform:org.eclipse.equinox.registry:jar:3.9.0 -> org.eclipse.platform:org.eclipse.equinox.common:jar:[3.13.0,4.0.0)", with Eclipse 2022-12 M1)

**Comment by mickaelistria** [comment]
I don't think this error is a blocking one. Have you tried a local build first and then updating projects?

**Comment by tomaswolf** [comment]
> I don't think this error is a blocking one.

The Oomph setup problem _is_ blocking for me as far as providing a fix is concerned. The wrong preference file being written of course is not a "blocker", but I don't think I claimed it was. It _is_ a major nuisance though in a team where team members work on different OSes, one of them being Windows.

> Have you tried a local build first and then updating projects?

I tried various things, including clean build and updating projects. No success.

**Comment by kwin** [comment]
@tomaswolf The problem with Tycho 3 is transitive dependencies using version ranges (https://github.com/eclipse-tycho/tycho/issues/1464). Try resolving from Maven Central directly instead of using a proxy repo manager with a fresh local repo. Hopefully that helps

**Comment by tomaswolf** [comment]
Thanks @kwin; see https://github.com/eclipse-tycho/tycho/issues/1464#issuecomment-1284429924. Your pointer to that tycho issue helped indeed. Looks like a tycho bug to me. My solution to that particluar tycho problem: put the missing JAR into the local repo directly.

**Comment by laeubi** [comment]
> Looks like a tycho bug to me

Then please provide an [integration-test](https://github.com/eclipse/tycho/wiki#providing-an-integration-test) to demonstrate the issue. We have hundreds of build running with Tycho 3.0 and don't see an issue, also Tycho itself is build using Tycho 3.0 also without an issue.

**Comment by tomaswolf** [comment]
> Then please provide an [integration-test](https://github.com/eclipse/tycho/wiki#providing-an-integration-test) to demonstrate the issue. We have hundreds of build running with Tycho 3.0 and don't see an issue, also Tycho itself is build using Tycho 3.0 also without an issue.

Don't have time for that. I'm working on other projects. If you want to analyze the issue, make sure you use a local .m2 repo that does not have any 3.16.* version of org.eclipse.platform:org.eclipse.equinox.common, then provision a fresh Eclipse via Oomph (Eclipse for Committers, latest, choose only m2e). Provisioning will work, setting up the workspace will fail during build.
